pages/chat/index.js

```diff
@@ -0,0 +1,186 @@
+import { useSession } from "next-auth/react";
+import { useState, useEffect, useRef } from "react";
+import Pusher from 'pusher-js';
+import { mutate } from "swr";
+import styled, { css } from "styled-components";
+import SendMessage from "@/components/Chat/SendMessage";
+
+export default function Chat() {
+  const messageLastRef = useRef(null)
+  const [chats, setChats] = useState([]);
+  const [messageToSend, setMessageToSend] = useState("");
+  const { data: session } = useSession();
+  const userName = session?.user.name;
+
+  useEffect(() => {
+    const pusher = new Pusher( process.env.NEXT_PUBLIC_PUSHER_KEY , {
+      cluster: 'eu'
+    });
+
+    const channel = pusher.subscribe('chat');
+
+    channel.bind("chatMessages", function (data) {
+
+      if (!data.receiver) {
+        setChats((prevState) => [
+          ...prevState,
+          { sender: data.sender, message: data.message },
+        ]);
+      }
+    });
+
+    return () => {
+      pusher.unsubscribe("chat");
+    };
+
+  }, []);
+
+  useEffect(() => {
+    messageLastRef.current?.scrollIntoView({ behavior: "smooth" });
+  },[chats])
+
+  if (!userName) return <h2>Loading ...</h2>
+
+  async function handleSubmit(event) {
+    event.preventDefault();
+    //const newMessage = event.target[0].value
+
+    try {
+      await fetch("/api/chat", {
+        method: "POST",
+        headers: {
+          "Content-Type": "application/json",
+        },
+        body: JSON.stringify({ message: messageToSend, sender: userName }),
+      });
+
+      await mutate;
+
+    } catch (error) {
+      console.error("Fehler beim Senden der Nachricht:", error);
+    }
+
+    setMessageToSend('')
+  }
+
+  return (
+    <StyledChat>
+      <UserList>
+        <h4>Welcome</h4>
+        <p>{ userName }</p>
+      </UserList>
+      <ChatContent>
+        <ChatMessages>
+          {chats.map((chat, id) => (
+            <Message key={ id } variant={chat.sender === userName ? 'current' : 'other' }>
+              <Sender>{ chat.sender }</Sender>
+              <Bubble>{ chat.message }</Bubble>
+            </Message>
+          ))}
+          <div ref={messageLastRef}></div>
+        </ChatMessages>
+        <SendMessage
+          message={messageToSend}
+          handleMessageChange={(event) => setMessageToSend(event.target.value)}
+          handleSubmit={(event) => {
+          handleSubmit(event)
+        }}/>
+      </ChatContent>
+    </StyledChat>
+  );
+}
+
+const StyledChat = styled.div`
+  display: flex;
+  align-items: stretch;
+  width: 100vw;
+  height: 100vh;
+  position: absolute;
+  top: 0;
+  left: 0;
+  right: 0;
+  bottom: 0;
+`;
+
+const UserList = styled.div`
```

Comment:
There are some issue here:
- the naming is confusing - a list with only one item?
- the usage of `h4` and a `p` tag is not really semantic - in how far is "Welcome, " a heading for the chat? and why is the user name not part of it as well?

Reply:
Only placeholder, in future case this should be a user list, but its my last day of capstone phase